remoteappmanager/application.py

```diff
@@ -181,11 +191,43 @@ def _reverse_proxy_init(self):
             self._config.proxy_api_url
         ))
 
+    def _db_init(self):
+        """Initializes the database connection."""
+        self.db = orm.Database(self.config.db_url)
+
+    def _user_init(self):
+        """Initializes the user at the database level."""
+        session = self.db.create_session()
+
+        with transaction(session):
+            try:
+                user = session.query(orm.User).filter_by(
+                    name=self.config.user).one_or_none()
+            except MultipleResultsFound:
+                self.log.error("Multiple results found when "
+                               "querying for username {}. This is supposedly "
+                               "impossible because the username should be a "
+                               "unique key by design.".format(
+                                   self.config.user))
+                raise
+
+            if user is None:
+                user = orm.User(name=self.config.user)
+                session.add(user)
```

Comment:
> Do we definitely want to add any unfound user to the databas

I have no idea. It's a matter of taste I guess? We need to extract the user anyway, and if we set the spawner to create the user, then we need to import the database and the orm objects from the spawner...

I personally think that once you spawn this process, you are already authenticated, and you either are already in the database, or if you are not, you should be. I would go with the solution that is less entangled.

@itziakos opinion?